sor/src/main/java/com/bazaarvoice/emodb/sor/db/astyanax/CqlDataWriterDAO.java

```diff
@@ -0,0 +1,211 @@
+        _readerDao = checkNotNull(readerDao, "readerDao");
+        _astyanaxWriterDAO = checkNotNull(delegate, "delegate");
+        _fullConsistencyTimeProvider = checkNotNull(fullConsistencyTimeProvider, "fullConsistencyTimeProvider");
+        _rawConsistencyTimeProvider = checkNotNull(rawConsistencyTimeProvider, "rawConsistencyTimeProvider");
+        _auditStore = checkNotNull(auditStore, "auditStore");
+        _changeEncoder = checkNotNull(changeEncoder, "changeEncoder");
+        _updateMeter = metricRegistry.meter(getMetricName("updates"));
+        _oversizeUpdateMeter = metricRegistry.meter(getMetricName("oversizeUpdates"));
+        _deltaBlockSize = deltaBlockSize;
+        _deltaPrefix = deltaPrefix;
+        _deltaPrefixLength = deltaPrefix.length();
+    }
+
+    private String getMetricName(String name) {
+        return MetricRegistry.name("bv.emodb.sor", "CqlDataWriterDAO", name);
+    }
+
+    @Override
+    public long getFullConsistencyTimestamp(Table table) {
+        return _astyanaxWriterDAO.getFullConsistencyTimestamp(table);
+    }
+
+    @Override
+    public long getRawConsistencyTimestamp(Table table) {
+        return _astyanaxWriterDAO.getRawConsistencyTimestamp(table);
+    }
+
+    @Override
+    public void updateAll(Iterator<RecordUpdate> updates, UpdateListener listener) {
+        _astyanaxWriterDAO.updateAll(updates,listener);
+    }
+
+    @Override
+    public void compact(Table tbl, String key, UUID compactionKey, Compaction compaction, UUID changeId, Delta delta, Collection<UUID> changesToDelete, List<History> historyList, WriteConsistency consistency) {
+        checkNotNull(tbl, "table");
+        checkNotNull(key, "key");
+        checkNotNull(compactionKey, "compactionKey");
+        checkNotNull(compaction, "compaction");
+        checkNotNull(changeId, "changeId");
+        checkNotNull(delta, "delta");
+        checkNotNull(changesToDelete, "changesToDelete");
+        checkNotNull(consistency, "consistency");
+
+        AstyanaxTable table = (AstyanaxTable) tbl;
+        for (AstyanaxStorage storage : table.getWriteStorage()) {
+            DeltaPlacement placement = (DeltaPlacement) storage.getPlacement();
+            CassandraKeyspace keyspace = placement.getKeyspace();
+
+            ByteBuffer rowKey = storage.getRowKey(key);
+
+            // Should synchronously write compaction and then delete deltas
+            writeCompaction(rowKey, compactionKey, compaction, consistency, placement, keyspace, tbl, key);
+
+            deleteCompactedDeltas(rowKey, consistency, placement, keyspace, changesToDelete, historyList, tbl, key);
+        }
+    }
+
+    private void insertBlockedDeltas(BatchStatement batchStatement, DeltaTableDDL tableDDL, ConsistencyLevel consistencyLevel, ByteBuffer rowKey, UUID changeId, ByteBuffer encodedDelta, int deltaSize) {
+        int numBlocks = (deltaSize + _deltaBlockSize - 1) / _deltaBlockSize;
```

Comment:
No, this expression will always result in a result of 1 or higher. It is essentially doing Math.ceil(deltaSize / deltaBlockSize) without any typecasting